Common/Extensions/StreamExtensions.cs

```diff
@@ -453,6 +459,59 @@ bool System.IAsyncResult.IsCompleted
 
         }
 
+        /// <summary>
```

Comment:
Supports more lazy evaultations?

Reply:
The problem here is that we do smth like this:

```
using (DownloadAdapter d = new DownloadAdapter())
    return d;
```

So returned `d` is disposed by `using` block and contains invalid values.

Actually we need to extend lifetime (aka not `Dispose`) `DownloadAdapter` till its `DownloadAdapater.ResponseOutputStream` used because `DownloadAdapter.Dispose()` do `DownloadAdapater.ResponseOutputStream.Dispose()`.

So `DownloadAdapter` &&  `DownloadAdapater.ResponseOutputStream` should be both not Disposed to use  `DownloadAdapater.ResponseOutputStream`. `DownloadStream` wraps and stores `DownloadAdapter`, so you cane safely use `DownloadAdapater.ResponseOutputStream` and when you do `DownloadStream.Dispose()` both   `DownloadAdapter` and `DownloadAdapater.ResponseOutputStream` are disposed.